src/views/Tweets.vue

```diff
@@ -1,33 +1,35 @@
 <template lang="pug">
   .container
     Header
-    p Hello tweets
-    b-form-input#input(type="text" v-model="message")
-    b-btn(@click="search") Search 
-    b-btn(@click="") Check Opinion <!-- feature for seeing the general opinion for all tweets found in a search -->
-    b-list-group
-      b-list-group-item(v-for="(node, index) in tweets" :key="index").flex-column.align.items.start
-        .d-flex.w-100.justify-content-between
-          h5.mb-1 {{node['user']}}
-          small 3 days ago
-        p.mb-1 {{node['text']}}
+    LoadingIcon(v-if="!done")
+    TweetList(:tweets="tweets" v-if="done")
 </template>
 
 <script lang="ts">
-import { Component, Vue, Watch } from 'vue-property-decorator';
+import { Component, Vue } from 'vue-property-decorator';
 import Header from '@/components/Header.vue';
+import LoadingIcon from '@/components/LoadingIcon.vue';
+import TweetList from '@/components/TweetList.vue';
 
 @Component({
   components: {
-    Header
-  },
+    Header,
+    LoadingIcon,
+    TweetList
+  }
 })
+
 export default class Tweets extends Vue {
   twitter_endpoint : string = `https://us-central1-saot-217513.cloudfunctions.net/tweets`;
   tweets : Array<Object> = [];
+  done : boolean = false;
 
   created () {
-    this.search(this.$route.params.query);
+    if (!this.$route.params.query) {
```

Comment:
Consider saving this to a temporary variable like `let query = this.$route.params.query;` to avoid duplication.

Reply:
- fixed fd74b218ad0d36adbc9ab08a663495dee5a2fb62